src/QsFmt/Parser/QSharpParser.g4

```diff
@@ -157,8 +157,10 @@ statement
     | 'until' expression (';' | 'fixup' scope) # UntilStatement
     | 'within' scope # WithinStatement
     | 'apply' scope # ApplyStatement
-    | ('use' | 'using') (qubitBinding | '(' qubitBinding ')') (';' | scope) # UseStatement
-    | ('borrow' | 'borrowing') (qubitBinding | '(' qubitBinding ')') (';' | scope) # BorrowStatement
+    | use=('use' | 'using') (binding=qubitBinding | openParen='(' binding=qubitBinding closeParen=')') semicolon=';' # UseStatement
+    | use=('use' | 'using') (binding=qubitBinding | openParen='(' binding=qubitBinding closeParen=')') body=scope # UseBlock
+    | borrow=('borrow' | 'borrowing') (binding=qubitBinding | openParen='(' binding=qubitBinding closeParen=')') semicolon=';' # BorrowStatement
+    | borrow=('borrow' | 'borrowing') (binding=qubitBinding | openParen='(' binding=qubitBinding closeParen=')') body=scope # BorrowBlock
```

Comment:
nit: To match the change to the syntax tree nodes, you could reduce these 4 rules down to 2 or 1 if you wanted.

Reply:
merged into 1 rule